app/views/layouts/application.html.erb

```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>PrivateEvents</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+  </head>
+
+  <body>
+    <%= link_to 'All Events', events_path %>
+    <%= link_to 'Users', users_path %>
+
+    <div class="users-session">
+      <% if current_user %>
```

Comment:
please remove all the logic from your views and put it inside the view helpers `app/helpers/*_helper.rb`